experiments/202005-bazel-rules-typescript/shipping-container-types/types.d.ts

```diff
@@ -0,0 +1,13 @@
+declare namespace shippingContainer {
```

Comment:
what's the point of namespaces in typescript? Just in case you want to re-use class names?

Reply:
Yeah good question. They're supposedly passé now but I think that's just for regular typescript. According to my reading and understanding, it's "best practice" with ts defs to put them in namespaces such that you minimize pollution of the global namespace. And "modules" in definitions are different than modules in regular ts files - and I don't think you're supposed to use them (?).

I am mildly confused by all of this. I'm maybe 60% sure that this is the right thing to do, as a best-practice matter.

Reply:
Yeah good question. They're supposedly passé now but I think that's just for regular typescript. According to my reading and understanding, it's "best practice" with ts defs to put them in namespaces such that you minimize pollution of the global namespace. And "modules" in definitions are different than modules in regular ts files - and I don't think you're supposed to use them (?).

I am mildly confused by all of this. I'm maybe 60% sure that this is the right thing to do, as a best-practice matter.

Reply:
Yeah good question. They're supposedly passé now but I think that's just for regular typescript. According to my reading and understanding, it's "best practice" with ts defs to put them in namespaces such that you minimize pollution of the global namespace. And "modules" in definitions are different than modules in regular ts files - and I don't think you're supposed to use them (?).

I am mildly confused by all of this. I'm maybe 60% sure that this is the right thing to do, as a best-practice matter.